src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1494,14 +1517,28 @@ private IEnumerable<TypeInferenceInfo> InferTypeInRecursivePattern(RecursivePatt
             }
 
             private IEnumerable<TypeInferenceInfo> GetPatternTypes(PatternSyntax pattern)
-                => pattern switch
+            {
+                return pattern switch
                 {
                     ConstantPatternSyntax constantPattern => GetTypes(constantPattern.Expression),
-                    DeclarationPatternSyntax declarationPattern => GetTypes(declarationPattern.Type),
                     RecursivePatternSyntax recursivePattern => GetTypesForRecursivePattern(recursivePattern),
-                    _ => SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>(),
+                    _ => GetPatternTypesFromOperation(pattern),
                 };
 
+                IEnumerable<TypeInferenceInfo> GetPatternTypesFromOperation(PatternSyntax pattern)
```

Comment:
i don't see any value in a local function here.  Why not just a normal switch, and then this local function will be what you do in the default case?

Reply:
Inlined in 4b67cba2